### ISSUE TITLE
[CAIA a11y Research - PILOT]: Sitewide Public Websites, Discharge Upgrade Wizard (Oct. 2024)

### ISSUE THREAD

**Issue by sara-amanda** [department-of-veterans-affairs/va.gov-team#94866]
> [!IMPORTANT]
> - **TICKET STATUS:** `COMPLETED`
> - **PILOT DATE:**  10/16 1-2 p.m. CT (2-3 p.m. EST)
> - **TYPES OF ASSISTIVE TECH:**  **Mobile Testing**: `iPhone / VoiceOver` with `Safari`
> - **TESTING ENVIRONMENT:** [Staging](https://staging.va.gov/discharge-upgrade-instructions/introduction)

# CAIA A11y Support
This is a child ticket of the research ticket [94815](https://github.com/department-of-veterans-affairs/va.gov-team/issues/94815) - created to track work, completed from this pilot session.
> [!TIP]
> **DEFINITION OF DONE**: Refer to the task list in this ticket, for actionable items.

## CAIA Assignee(s)

<details><summary>Toggle to view CAIA info</summary>

- **Pilot**: @SarahKay8  
- **Co-pilot**: @coforma-jamie 

</details>

## Team Info

<details><summary>Toggle to view team info</summary>

- Team name: Sitewide Public Websites
- OCTO-DE product owner: Michelle Middaugh (PO), Amanda Klausmeier (VA UX Lead)
- Product manager: Fran Cross
- Designer: Jordan Wood
- FE engineer: Chris Kim
- Product/team Slack channel: #sitewide-public-websites

</details>

## Research Goals

<details><summary>Toggle to view team research goals</summary>

> 1. Veterans can understand how to edit an answer that introduces branching logic and navigate back to the review page.
> 2. Veterans can understand why they may be asked additional questions when editing an answer to a particular question.
> 3. After editing a question, updates to the review page are clear and easy for Veterans to understand.
> 4. Veterans can navigate through the entire flow using the subtask pattern with ease.
> 5. Veterans that use assistive technology, like screen readers or screen magnifier technology, can navigate through user flows with ease.

> _We will conduct remote moderated usability testing to observe research participants using the Discharge Upgrade Wizard in a live coded environment (staging or production) with the updated sub task pattern. This will allow us to use the current branching logic and our experimental edit flow pattern._

</details>

## Description

<details><summary>Toggle to view the description</summary>

**A CAIA a11y will participate as a user in a pilot of this research study.**

- **Researcher**: The VFS team researcher will test their prototype and conversation guide by conducting the pilot session with a CAIA A11Y.
- **User**: The CAIA A11Y team member will act as a real user in the pilot session.
- **Assistive Tech**: The CAIA A11Y team member will follow the prompts given by the VFS Researcher, while using and demonstrating how the behavior of a `screen reader` would behave.

### Pilot Session
- **Date:** 10/16/2024 1-2 p.m. Central (2-3 p.m. EST)
- **Prototype**: [Staging](https://staging.va.gov/discharge-upgrade-instructions/introduction)
- [Research Plan](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/veteran-military-records/discharge-upgrade-wizard/research/2024-Review-Flow/research-plan.md) 
- [Research Convo Guide](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/veteran-military-records/discharge-upgrade-wizard/research/2024-Review-Flow/conversation-guide-at.md)
</details>

### Assistive Tech Options

> [!TIP]
> **DEVICE SUPPORT DURING RESEARCH SESSIONS**
> - **Available AT**: `SK` will test using `VoiceOver` on `Safari` and `iPhone`
> - **Additional Specifications:** If there is a particular `test user`, `browser` or `operating system` you would like us to use, please let us know in advance via this ticket's comments, tagging the assignee - or reach out to @coforma-terry in the CAIA Slack Channel: [#sitewide-content-accessibility-ia ](https://dsva.slack.com/archives/C01K37HRUAH)

## Definition of Done

```[tasklist]
### A11y Tasks
- [x] **Schedule CAIA a11ys** SK and Jamie
- [x] **Share link to [Assistive Tech Pilot Guide](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/teams/CAIA/accessibility/assistive-tech-pilot-guide.md) with team, if they haven't reviewed it yet**
- [x] **Determine date and time for pilot**
- [x] **Update CAIA A11y Capacity Calendar:** Add invite for pilot, when scheduled
- [x] **Review research** [conversation guide](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/veteran-military-records/discharge-upgrade-wizard/research/2024-Review-Flow/conversation-guide-at.md)
- [x] [Request AT device and tech to be used in sessions](LINK)
- [x] [Receive AT device and tech from Team in research ticket](LINK) to be used in sessions to prep for pilot
- [ ] **Review prototype** in `Staging`
- [ ] **Run pilot** with VFS Researcher
- [ ] **Optional: Note Taking:** [Notes doc](https://docs.google.com/document/d/13FAzYKrnwOT4n7MrsEcqOkfI5M8WfBKDwc6Veh87xco/edit?usp=sharing)
- [x] Share findings with team from Pilot doc.
```

**Comment by coforma-jamie** [comment]
## Pilot notes

- Overall, the flow is good!
- Technical issues:
  - iPhone + VO + Safari is reading radio buttons twice. this is a component issue and Rachel is going to look into it
  - “What year were you discharged from the military?” - visual focus is hanging on the Back button. (The black focus indicator here is from VoiceOver - it's where true focus is - but the yellow visual indicator is still active on Back button)
![Image](https://github.com/user-attachments/assets/735f8f9b-d398-4223-b7fd-b8e54add059b)
- Usability issues:
  - Consider changing Year to an input field (rather than a dropdown)
  - Throughout, but especially on the “Your Steps for Upgrading Your Discharge,” there is a LOT of content. Recommend working with CAIA to trim it down as much as possible.
    - Also, on that page, if the form the Vet needs to fill out isn’t available online (PDF only), say that near the top of the page so they know what to expect.

cc @rachaelpenfil

**Comment by SarahKay8** [comment]
For radio button repetitive announcement:

The repetition issue with VoiceOver is likely due to redundant labeling between the va-radio-option component and the nested HTML structure. VoiceOver might be reading both the label attribute on the va-radio-option component and the inner <label> element, causing it to announce the label twice.

#To Fix VoiceOver Repetition

Remove the label attribute from the va-radio-option component and rely solely on the inner <label> element.

<va-radio-option checked="false" data-testid="va-radio-option" name="group" value="I was discharged due to my sexual orientation (including under the Don’t Ask, Don’t Tell policy)." uswds="true" class="hydrated">
    <div class="usa-radio" wiid="N537">
        <input class="usa-radio__input" type="radio" name="group" value="I was discharged due to my sexual orientation (including under the Don’t Ask, Don’t Tell policy)." id="groupI-was-discharged">
        <label class="usa-radio__label" for="groupI-was-discharged">I was discharged due to my sexual orientation (including under the Don’t Ask, Don’t Tell policy).</label>
    </div>
</va-radio-option>

#Second option 

Keep  the label attribute on the va-radio-option component and remove the inner <label> element entirely. This approach depends on whether the va-radio-option component automatically maps the label for accessibility.
